src/WebJobs.Script/Diagnostics/ConditionalTraceWriter.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
```

Comment:
@mathewc this is the conditional trace writer I mentioned. I didn't add the property check support we've discussed as it wasn't currently needed (opted to go with the additional trace writer for now), but we can add that if we decide to go that way. There are some side effects we can discuss if we decide to go down that path.